pcdsdevices/utils.py

```diff
@@ -687,3 +690,151 @@ def post_ophyds_to_elog(objs, allow_child=False, hutch_elog=None):
+    """
+    start_with = _normalize_reorder_list(cls, start_with)
+    end_with = _normalize_reorder_list(cls, end_with)
+    for cpt_name in reversed(start_with):
+        cls._sig_attrs.move_to_end(cpt_name, last=False)
+    for cpt_name in end_with:
+        cls._sig_attrs.move_to_end(cpt_name, last=True)
+
+
+def _normalize_reorder_list(
+    cls: type,
+    cpts_or_names: Optional[List[Union[str, Cpt]]],
+) -> List[str]:
+    """
+    Simplify the user's variable arguments for the component reordering.
+    """
+    if cpts_or_names is None:
+        return []
+    reverse_map = {cpt: name for name, cpt in cls._sig_attrs.items()}
+    output = []
+    for obj in cpts_or_names:
+        if isinstance(obj, Cpt):
+            output.append(reverse_map[obj])
+        else:
+            output.append(obj)
+    return output
+
+
+def move_subdevices_to_start(cls: type, subdevice_cls: type = Device):
+    """
+    Arrange the component order of a device class to put subdevices first.
+
+    This can be useful to bring e.g. all the motors to the top for the
+    typhos screen.
+
+    The relative ordering of subdevices is preserved.
+
+    Parameters
+    ----------
+    cls : type
+        The Device subclass that we'd like to rearrange the order of.
+    subdevice_cls: type, optional
+        A specific class type to move to the front. If omitted, all device
+        subclasses will be moved.
+    """
+    device_names = []
+    for name, cpt in cls._sig_attrs.items():
+        if issubclass(cpt.cls, subdevice_cls):
+            device_names.append(name)
+    reorder_components(cls, start_with=device_names)
+
+
+def sort_components_by_name(cls: type, reverse: bool = False):
+    """
+    Arrange the component order of a device class in alphabetical order.
+
+    This can be useful as a first step before bringing specific components
+    to the top of the queue for the typhos screen.
+
+    Parameters
+    ----------
+    cls : type
+        The Device subclass that we'd like to rearrange the order of.
+    reverse : bool, optional
+        Set to True to sort in descending order instead.
+    """
+    alphabetical = list(sorted(cls._sig_attrs, reverse=reverse))
+    reorder_components(cls, start_with=alphabetical)
+
+
+def sort_components_by_kind(cls: type):
+    """
+    Arrange the component order of a device class in kind order.
+
+    Kind order is hinted > normal > config > omitted.
+
+    This can be useful because typically the higher kind classes
+    are more important, and therefore should be higher up on the
+    typhos screen.
+
+    The relative ordering of subdevices within a kind is preserved.
+
+    Parameters
+    ----------
+    cls : type
+        The Device subclass that we'd like to rearrange the order of.
+    """
+    hinted = []
+    normal = []
+    config = []
+    omitted = []
+    for name, cpt in cls._sig_attrs.items():
+        if cpt.kind is Kind.hinted:
```

Comment:
`ophyd.Kind` is a flag, so it could be several of these (even if it's not typically the case).
That is, it could be `hinted | config` (implied `normal` due to how the bitmask is set up) which would be entirely valid.

Reply:
This is a holdover from my willful ignoring of the bitflag behavior, which I've never found a use for.

Ok, so the bitflag choices are the following (and sums thereof):
```
    omitted = 0b000
    normal = 0b001
    config = 0b010
    hinted = 0b101
```

What would the full ordering be then? How do I classify a signal that is `hinted & omitted`, or something like `normal & config`? Do I simply take the highest order component, e.g. all signals with `hinted` on them are the first group? Do I have to sort within the `hinted` group too, like is `hinted/normal` before `hinted/config`? Is `normal/config` before or behind `normal`?

I didn't have any good answers to this while implementing the class. If the numbers here were in order we could simply sort on the kind int number as-is, but they are not (numerically, hinted > config > normal > omitted)

I'm tempted to just group all "hinted" signals without regard for in-hinted ordering, then all "normal" signals (with no hint flag), then all "config" signals (with no hinted or normal flags), and then the omitted signals at the end.

Reply:
> I'm tempted to just group all "hinted" signals without regard for in-hinted ordering, then all "normal" signals (with no hint flag), then all "config" signals (with no hinted or normal flags), and then the omitted signals at the end.

I think that's the appropriate path forward here - I similarly have no good answer to your questions above. 
The change would then just be to the effect of replacing `kind is Kind.hinted` with `(kind & kind.Hinted) == kind.Hinted`

Reply:
Some additional nuance, a lot of the combinations alluded to above don't actually make sense because the `hinted` flag includes the `normal` flag already. If we limit ourselves to `|` operations, the only possible values are:
```
0b101 == hinted
0b001 == normal
0b010 == config
0b000 == omitted
0b111 == hinted | config
0b011 == normal | config
```
The remaining two bit combinations are:
```
0b110 == ~ normal
0b100 == hinted & ~ normal
```
So this does mean that it's feasible to do a full top to bottom ordering, there are only 6 options. I think I probably still won't though.